src/platforms/stm32/traceswoasync_f723.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * This file is part of the Black Magic Debug project.
+ *
+ * Based on work that is Copyright (C) 2017 Black Sphere Technologies Ltd.
+ * Copyright (C) 2017 Dave Marples <dave@marples.net>
+ * Portions (C) 2020-2021 Stoyan Shopov <stoyan.shopov@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.	 If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* This file implements capture of the TRACESWO output using ASYNC signalling.
+ *
+ * ARM DDI 0403D - ARMv7M Architecture Reference Manual
+ * ARM DDI 0337I - Cortex-M3 Technical Reference Manual
+ * ARM DDI 0314H - CoreSight Components Technical Reference Manual
+ */
+
+/* TDO/TRACESWO signal comes into the SWOUSART RX pin.
+ */
+
+#include "general.h"
+#include "usb.h"
+#include "traceswo.h"
+
+#include <libopencmsis/core_cm3.h>
+#include <libopencm3/cm3/nvic.h>
+#include <libopencm3/stm32/timer.h>
+#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/usart.h>
+#include <libopencm3/stm32/dma.h>
+
+static volatile uint32_t w; /* Packet currently received via UART */
+static volatile uint32_t r; /* Packet currently waiting to transmit to USB */
+/* Packets arrived from the SWO interface */
+static uint8_t trace_rx_buf[NUM_TRACE_PACKETS * TRACE_ENDPOINT_SIZE];
+/* Packet pingpong buffer used for receiving packets */
+static uint8_t pingpong_buf[2 * TRACE_ENDPOINT_SIZE];
+/* SWO decoding */
+static bool decoding = false;
+
+void trace_buf_drain(usbd_device *dev, uint8_t ep)
+{
+	static volatile char inBufDrain;
```

Comment:
This should be changed to use C11 atomics instead of compiler built-ins please.

https://en.cppreference.com/w/c/atomic/atomic_flag provides the reference required for this particular pattern. You can also find it in use in https://github.com/blackmagic-debug/blackmagic/blob/c6890e120d9bc1761095fc90820c646f22e8af52/src/platforms/stm32/traceswoasync.c#L57